src/Arcus.BackgroundJobs.KeyVault/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -20,6 +21,135 @@ namespace Microsoft.Extensions.DependencyInjection
+        /// </remarks>
+        /// <param name="services">The services collection to add the job to.</param>
+        /// <param name="topicName">The name of the Azure Service Bus Topic to process the received event.</param>
+        /// <param name="subscriptionNamePrefix">The name of the Azure Service Bus subscription that will be created to receive Key Vault events.</param>
+        /// <param name="serviceBusNamespace">The Service Bus namespace to connect to. This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <paramref name="serviceBusNamespace"/>, the <paramref name="topicName"/>, or the <paramref name="subscriptionNamePrefix"/> is blank.
+        /// </exception>
+        public static IServiceCollection AddAutoInvalidateKeyVaultSecretUsingManagedIdentityBackgroundJob(
+            this IServiceCollection services,
+            string topicName,
+            string subscriptionNamePrefix,
+            string serviceBusNamespace)
+        {
+            Guard.NotNull(services, nameof(services), "Requires a set of services to add the CloudEvents background job to");
+            Guard.NotNullOrWhitespace(topicName, nameof(topicName), "Requires a non-blank Azure Service Bus Topic name to identity the Azure Service Bus entity");
+            Guard.NotNullOrWhitespace(subscriptionNamePrefix, nameof(subscriptionNamePrefix), "Requires a non-blank subscription name of the Azure Service Bus Topic subscription, to receive Key Vault events");
+            Guard.NotNullOrWhitespace(serviceBusNamespace, nameof(serviceBusNamespace), "Requires a non-blank fully qualified namespace for the Azure Service Bus Topic");
+
+            return AddAutoInvalidateKeyVaultSecretUsingManagedIdentityBackgroundJob(services, topicName, subscriptionNamePrefix, serviceBusNamespace, clientId: null);
+        }
+
+        /// <summary>
+        /// Adds a background job to the <see cref="IServiceCollection"/> to automatically invalidate cached Azure Key Vault secrets.
+        /// </summary>
+        /// <remarks>
+        ///     Make sure that the application has the Arcus secret store configured correctly.
+        ///     For on the Arcus secret store: <a href="https://security.arcus-azure.net/features/secret-store" />.
+        /// </remarks>
+        /// <param name="services">The services collection to add the job to.</param>
+        /// <param name="topicName">The name of the Azure Service Bus Topic to process the received event.</param>
+        /// <param name="subscriptionNamePrefix">The name of the Azure Service Bus subscription that will be created to receive Key Vault events.</param>
+        /// <param name="serviceBusNamespace">The Service Bus namespace to connect to. This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="clientId">
```

Comment:
The clientId parameter actually represents a user-assigned managed identity ?  Shouldn't we then give it another name ?
If I see clientId, I always think that I need a corresponding client-secret.

Reply:
This is actually a copy-paste from the actual `<param/>` tag in the `ManagedidentityCredential`-stuff, so we used `clientId` in all managed identity-related stuff across Arcus.

Reply:
Looks like I've ignored this PR for a while.  Or maybe it didn't pop up in my assigned PR's to review list :?